src/Modal/Modal.tsx

```diff
@@ -172,22 +172,26 @@ export const Modal: React.FC<Props> = ({
 
   /**
    * Framer motion component to render. The type will be taken from the `as`
-   * prop
+   * prop.
+   *
+   * We have to strip the type because we'll get an error about the union being
+   * too complex to model because `motion` has over 100 options.
    */
-  const MotionComponent = motion[type];
+  const MotionComponent = motion[type] as React.ComponentType<any>;
```

Comment:
NDB: would unknown work here? (and typing the variable instead of the assertion?)
```suggestion
  const MotionComponent: React.ComponentType<unknown> = motion[type];
```

Reply:
Thinking about this a little more I looked into the union type that I removed (`motion`). They implement `MotionProps` that I can use here and then the motion props will be typed correctly!

Reply:
Also, 👍 for typing the variable instead of the assertion.

Reply:
oohh thats even better! :)